proof/refine/ARM/Invariants_H.thy

```diff
@@ -3393,4 +3393,50 @@ add_upd_simps "invs' (gsUntypedZeroRanges_update f s)"
   (obj_at'_real_def)
 declare upd_simps[simp]
 
+lemma neq_out_intv:
+  "\<lbrakk> a \<noteq> b; b \<notin> {a..a + c - 1} - {a} \<rbrakk> \<Longrightarrow> b \<notin> {a..a + c - 1}"
```

Comment:
I guess this lemma is pretty specific, but do we want to move it to `lib`? That would at least reduce duplication while we haven't arch-split `Refine`.